### ISSUE TITLE
Vanilla Headgear and purple squares

### ISSUE THREAD

**Issue by Hallanvaara** [AthenaSulisMinerva/CombatExtendedFastTrack#130]
Purple squares on pawns wearing vanilla headgear, apparently(?) game cant find textures. Modded headgear works for me. Forcing pawns drop headgear makes squares go away.

Happens on new and old saves. Tested on both CE and CE:FT. I dont think other mods can be responsible since headgear is vanilla and modded gear worked.

**Comment by BUTTERSKY11** [comment]
Check my issue point it might have the answer

**Comment by Hallanvaara** [comment]
Yeah, spend last 2 hours removing and adding mods and yes it seems best "fix" right now is to remove all mods that add gear. After removing 2 mods that add one set of armor in game. Purple squares disappeared.

Also noticed that even though squares disappeared the mod that adds visible hair even when wearing helmets etc. doesn't work anymore.

My personal fix is downloading older version of CE:FT and manually installing it and using it to run my save.

**Comment by BUTTERSKY11** [comment]
What version of Combat extended fast track is the functioning onee

**Comment by Theblessedwant** [comment]
a quick fix would be to remove the file as stated in my rimmu-nation fix. The dll also plays with head layers and conflicts with CE for some reason.

**Comment by BUTTERSKY11** [comment]
I would much rather have the older version where I can use prisoner's outfits and RPG Inventory

**Comment by sumghai** [comment]
Duplicate of #129, partially related to #127.

Usually, we would close newer duplicate tickets in favor of the first reported one, but I shut down the original as the discussion became uncivil. We'll continue from here.

Basically, as @Theblessedwant has determined, due to changes in how the upstream CE mod renders head apparel layers and faces, pretty much every other third-party mod that also modifies these layers are (for now) incompatible with CE (and CE:FT by extension). The original CE team has already stated that it will take time for all mod authors to update their mods to be compatible with the new system, so incompatibilities are always to be expected between major version jumps.

The Steam Workshop version of the mod will always only host the latest version of CE:FT for the convenience of the vast majority of users. For users who want to roll back to earlier versions (including those based on CE 1.5), they are available for manual download and installation from our [releases](https://github.com/AthenaSulisMinerva/CombatExtendedFastTrack/releases) page. 

In summary, this issue will resolve itself once other modders update their mods to be compatible with CE/CE:FT 1.6.

**Comment by Hallanvaara** [comment]
Yeah. One of those mods had that dll and by removing it squares disappeared. That mod hasn't been updated in a long time so I think I will try to learn how to make a patch for personally.

Still going to prefer older version of ce:ft because pawns without visible hair under headgear is IMHO ugly as f and makes pawns less individual. Too bad since I really liked how weapons looked after update and was looking towards seeing how changes affect gameplay

**Comment by sumghai** [comment]
@Hallanvaara - At the risk of veering off-topic, [Don't Shave Your Head!](https://github.com/NoImageAvailable/DontShaveYourHead/releases/) is compatible with CE 1.6, as it was also written by CE's NiA - he specifically updated it to work with the new layers.

**Comment by Hallanvaara** [comment]
Cool, have to check it. Also is there somewhere quick quide to patching so I could make little patch to starship troopers armor mod? I was thinking just copying values from power armor and maybe little tweaking

**Comment by sumghai** [comment]
For the time being, the [GitHub wiki for the original CE](https://github.com/NoImageAvailable/CombatExtended/wiki) contains useful guides on how to write patches. Eventually, the CE:FT will write its own wiki articles.

Now, let's get back on topic, shall we?

**Comment by Styrkyr** [comment]
It's  too rough. I had to roll back to 1.5. As game changingly good as CE:FT is, I couldn't justify disabling all the mods I have installed that are conflicting with it. Will update once other mod authors have updated as well. On a side note, have you guys noticed how lethal smoke inhalation is? Almost lost my colony last night to a ZZZZZT fire just from the inhalation. Anyways, keep up the good work modders, and thanks for a great mod.

**Comment by Dyrewulfe** [comment]
Referencing since this issue is pinned:
#143

**Comment by sumghai** [comment]
Thanks @Dyrewulfe - let's consolidate discussions there then.
